server/controllers/rules.js

```diff
@@ -0,0 +1,64 @@
+const model = require('../models')
+
+module.exports.findById = function (request, response) {
+  let scope = request.params.id ? { where: { id: request.params.id } } : {}
+  model.Rules.findAll(scope)
```

Comment:
hay un find by id, que te devuelve una sola rule, y no una colección de rules: http://docs.sequelizejs.com/manual/tutorial/models-usage.html

Además, cuando uno responde de una URL que siempre devuelve 1 instancia, no se da un 

rules: [only_rule]

sino

rule: only_rule

Reply:
Si devuelve dada una id, pero cual es el problema de eso? vos te referis a como ob tener un conjunto de reglas? esa funcion no esta en controllers, ya no no es necesaria, si miras el archivos server/routes/shipment_cost.js vas a ver como se obtiene el vector de reglas usando el la base de datos. Eso se puede refacoriar pero no es la idea de esta implementacion. Lo hare en un futuro branch.

Reply:
Pero vos hablas de un endpoint y me haces referencia a un controller. A que endpoint te referis?
porque el enpoint shipment-cost no devuelve rules. 
En cambio el endpoint rules, si llama a este controller y en ese caso si tenes razon, devuelve una sola instancia

Reply:
Listo, ya lo refactorice

Reply:
```
      if (JSON.stringify(scope) === JSON.stringify({})) {
        // here we return all the rules in a vector
        response.json({ success: true, rules: rules })
      } else {
        // here we return only one rule
        response.json({ success: true, rules: rules[0] })
      }
```

lo que digo es que
* se está usando el plural 'rules' cuando no estas dando un array, sino un solo rule
* se esta diciendo success true cuando no se encontró la instancia
* el status code por default (200: ok) se está usando cuando no se encontró la instancia, en lugar de 404 not found que es el que corresponde

en general si pegarle a una url no puede satisfacer lo que le piden se tira un status code de error

Reply:
lo de el plural tenes razon, pero aca te equivocas: `se esta diciendo success true cuando no se encontró la instancia` porque ese if pregunta si el get se hizo pasando un id o no, es decir, el if es para devolver todas las reglas en un vector o una sola regla (el objeto )

Reply:
Lo del plural no tengo manera de cambiarlo ya que es un atributo de la funcion dentro de then y recien en el if puedo conocer si quiero todos o uno solo

Reply:
Quizas el nombre de la funcion findById no este claro, pero si vos le pegas al route rules (get/id) devuelve la regla, pero si haces esto: (get/) devuelve todas

Reply:
aaaaah ya entiendo

pasa que para eso se suele definir en routes algo como

routes.get('ble/', findAll)
routes.get('ble/:id/', findByID)

y pasar ese if a 2 metodos distintos

pero nada, desde afuera es una api REST entonces, no es tan terrible

lo doy como resolved para no detener progreso por nimiedades, pero tene en cuenta que las urls del tipo GET /ble/2 suelen correr un metodo de controller distinto que GET /ble